<?xml version="1.0" encoding="utf-8"?>
<launch>
   <!-- run rviz-->
    <arg name="rviz"                default="true" />
    <!--     Name of the task file which defines the mpc problem settings-->
    <arg name="mpc_task_file"       default="task.info"/>
    <!-- Set path of robot description -->
    <arg name="description_file"    default="$(find perceptive_mpc)/urdf/mabi/mabi_mobile.urdf"/>
    <!-- Set rviz configuration -->
    <arg name="rvizconfig"          default="$(find perceptive_mpc)/config/demo.rviz"/>

    <!-- Load robot description -->
    <param name="robot_description" textfile="$(arg description_file)"/>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
        <param name="publish_frequency" value="100"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" if="$(arg rviz)"/>
    <!--node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"  output="screen"/>
    <node pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" name="joint_state_publisher_gui"  output="screen"/-->

    <node pkg="perceptive_mpc" name="kinematic_simulation" type="perceptive_mpc_kinematic_simulation" output="screen"> <!--launch-prefix="gdb -ex run \-\-args" -->
      <param name="mpc_task_file"           value="$(arg mpc_task_file)"/>
      <rosparam file="$(find perceptive_mpc)/config/kinematic_simulation_parameters.yaml"/>
    </node>

    <node name="interactive_pose_node" pkg="perceptive_mpc" type="interactive_marker.py" output="screen"/>
</launch>